src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -185,9 +185,12 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
         type: Show.type,
         description:
           "Most recent running Show or Fair booth the artwork is currently in, sorted by relevance",
-        resolve: async (artwork, {}, ctx) => {
+        resolve: async ({ show_ids }, {}, ctx) => {
+          if (!show_ids || show_ids.length === 0) {
+            return null
+          }
           const showOrFair = await await ctx.showsLoader({
-            artwork: artwork._id,
+            id: show_ids,
```

Comment:
querying in this style will likely be faster since we already have the pool of potential shows winnowed down with `show_ids`, vs. needing to query based on the artwork - https://github.com/artsy/gravity/blob/f7deb2d144075dc7d9a70a606868309c707dda0a/app/models/domain/partner_show.rb#L288